resource-managers/kubernetes/integration-tests/src/test/scala/org/apache/spark/deploy/k8s/integrationtest/SecretsTestsSuite.scala

```diff
@@ -105,7 +105,7 @@ private[spark] trait SecretsTestsSuite { k8sSuite: KubernetesSuite =>
       .withTTY()
       .exec(cmd.toArray: _*)
     // wait to get some result back
-    Thread.sleep(1000)
+    Thread.sleep(5000)
```

Comment:
Is there a better way to check for completion other than sleeping for some arbitrary period?

Because if there is you could use `eventually`, for example, and use an even longer timeout without penalizing cases where the command executes quickly.